tests/application/test_use_cases.py

```diff
@@ -95,8 +96,12 @@ def test_validate_data_submission(app_context, caplog):
     submission = app_context.submissions.get_by_id(1)
 
     class CustomMockTestTaskQueue:
-        def run_load_and_validate(self, path: str):
-            return MockCeleryTask(1)
+        def run_load_and_validate(
+            self, submissions: DataSubmissionRepository, submission_id: int, path: str
+        ):
+            return DataSubmissionReport(
+                overview=DataSubmissionReportOverview(feature_count=1)
+            )
 
     app_context._task_queue = CustomMockTestTaskQueue()
```

Comment:
Future thought - it might be helpful to have a way to provide a mock implementation to `create_app_context` (or maybe `create_test_app_context` - if this was done often.